spec/qos-classifier/02-integration_spec.lua

```diff
@@ -0,0 +1,181 @@
+local helpers = require "spec.helpers"
+local cjson = require "cjson"
+
+local PLUGIN_NAME = "qos-classifier"
+
+local config = {
+  termination = { 
+      status_code = 302,
+      header_name= "Location",
+      header_value="https://cred.club"
+  },
+  upstream_header_name = "X-QOS-CLASS",
+  node_count = {
+      http_timeout_in_ms = 15,
+      update_initial_delay_in_sec = 5,
+      initial = 1,
+      update_frequency_in_sec = 1
+  }
+}
+
+local function make_class(green, orange, red)
+  classes = {
+    class_1 = {
+      threshold = green,
+      header_value = "Green"
+    },
+    class_2 = {
+      threshold = orange,
+      header_value = "Orange"
+    },
+    class_3 = {
+      threshold = red,
+      header_value = "Red"
+    }
+  }
+  return classes
+end
+
+local function make_requests_to_upstream(host, start_time)
+  local client
+  while true do
+    client = helpers.proxy_client()
+    assert(client:get("/get", {
+      headers = { Host = host },
+    }))
+    ngx.sleep(0.01)
+    if(ngx.now() - start_time > 1.2) then break end
+  end
+end
+
+for _, strategy in helpers.each_strategy() do
+  describe(PLUGIN_NAME .. ": (integration) [#" .. strategy .. "]", function()
+    local bp, db
+  
+    lazy_setup(function()
+      bp, db = helpers.get_db_utils(strategy, nil, { PLUGIN_NAME })
+      local service = bp.services:insert()
+
+      local route1 = bp.routes:insert {
+        hosts   = {  "test1.com" },
```

Comment:
"test1.com", "test2.com" etc can be put in a variable for reusability and can have more intuitive values like "terminate.host", "green.host" etc

Reply:
Done